packages/vm/src/wasm_backend/store.rs

```diff
@@ -31,8 +31,23 @@ fn cost(_operator: &Operator) -> u64 {
     150_000
 }
 
-/// Creates an engine with the default compiler.
-pub fn make_engine(middlewares: &[Arc<dyn ModuleMiddleware>]) -> Engine {
+/// Creates an engine without a compiler.
+/// This is used to run modules compiled before.
+pub fn make_runtime_engine(memory_limit: Option<Size>) -> Engine {
+    let mut engine = Engine::headless();
+    if let Some(limit) = memory_limit {
+        let base = BaseTunables::for_target(&Target::default());
+        let tunables = LimitingTunables::new(base, limit_to_pages(limit));
+        engine.set_tunables(tunables);
+    }
+    engine
```

Comment:
Nit: This can be turned into a `match` without much effort.

Reply:
Not sure I inderstand. Why would a match be better here?

Reply:
Just style / avoid creating a variable.

Reply:
imho this situation is exactly the reason for `if let` to exist, so I'd keep it

Reply:
I guess what I didn't liked was the lone `engine` at the end. 🤷🏼‍♂️